flink-staging/flink-streaming/flink-streaming-connectors/flink-connector-rabbitmq/src/main/java/org/apache/flink/streaming/connectors/rabbitmq/RMQSource.java

```diff
@@ -90,10 +99,11 @@ public void run(SourceContext<OUT> ctx) throws Exception {
 			delivery = consumer.nextDelivery();
 
 			OUT result = schema.deserialize(delivery.getBody());
+			addId(ctx, delivery.getEnvelope().getDeliveryTag());

```

Comment:
There should be a lock scope around storing the ID and emitting the record. Have a look at the example code in the java docs of the MessageAcknowledingSourceBase class.